tests/test-cases/Group1-Docker-Commands/1-39-Docker-Stats.robot

```diff
@@ -0,0 +1,97 @@
+    ${vicMemory}=  Fetch From Left  ${vicMemory}  .
+    # get the latest memory value for the "stresser" vm
+    ${rc}  ${vmomiMemory}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc metric.sample -n 1 -json vm/*${short} mem.active.average | jq -r .Sample[].Value[].Value[0]
+    ${rc}  ${vmomiMemory}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc metric.sample -n 1 -json */%{VCH-NAME}/*${short} mem.active.average | jq -r .Sample[].Value[].Value[0]
+    Should Be Equal As Integers  ${rc}  0
+    Should Be True  ${vmomiMemory} > 0
+    # convert to percent and move decimal
+    ${percent}=  Evaluate  (${vmomiMemory}/2048000)*100
+    ${diff}=  Evaluate  ${percent}-${vicMemory}
+    # due to timing we could see some variation, but shouldn't exceed 5%
+    Should Be True  ${diff} < 5
+
+Stats No Stream All Containers
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} stats --no-stream -a
+    Should Be Equal As Integers  ${rc}  0
+    ${stress}=  Get Container ShortID  %{STRESSED}
+    ${stop}=  Get Container ShortID  %{STOPPER}
+    Should Contain  ${output}  ${stress}
+    Should Contain  ${output}  ${stop}
+
+Stats API Memory Validation
+    ${rc}  ${apiMem}=  Run And Return Rc And Output  curl -s -k -H "Accept: application/json" -H "Content-Type: application/json" -X GET https://%{VCH-IP}:%{VCH-PORT}/containers/%{STRESSED}/stats?stream=false | jq -r .memory_stats.usage
+    Should Be Equal As Integers  ${rc}  0
+    ${stress}=  Get Container ShortID  %{STRESSED}
+    ${rc}  ${vmomiMemory}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc metric.sample -n 1 -json vm/*${stress} mem.active.average | jq -r .Sample[].Value[].Value[0]
+    ${rc}  ${vmomiMemory}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc metric.sample -n 1 -json */%{VCH-NAME}/*${stress} mem.active.average | jq -r .Sample[].Value[].Value[0]
+    Should Be Equal As Integers  ${rc}  0
+    ${vmomiMemory}=  Evaluate  ${vmomiMemory}*1024
+    ${diff}=  Evaluate  ${apiMem}-${vmomiMemory}
+    ${diff}=  Set Variable  abs(${diff})
+    Should Be True  ${diff} < 1000
+
+Stats API CPU Validation
+    ${rc}  ${apiCPU}=  Run And Return Rc And Output  curl -s -k -H "Accept: application/json" -H "Content-Type: application/json" -X GET https://%{VCH-IP}:%{VCH-PORT}/containers/%{STRESSED}/stats?stream=false | jq -r .cpu_stats.cpu_usage.percpu_usage[0]
+    Should Be Equal As Integers  ${rc}  0
+    ${stress}=  Get Container ShortID  %{STRESSED}
+    ${rc}  ${vmomiCPU}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc metric.sample -json vm/*${stress} cpu.usagemhz.average | jq -r .Sample[].Value[0].Value[]
+    ${rc}  ${vmomiCPU}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc metric.sample -json */%{VCH-NAME}/*${stress} cpu.usagemhz.average | jq -r .Sample[].Value[0].Value[]
```

Comment:
can't do this var assignment like this, with esxi it will always get overwritten to null.